### PR TITLE
Fix duplicated rows in zkapp_account_precondition table

### LINKED ISSUE #14777: Archive Node: Duplicated entries in the `zkapp_account_preconditions` table
Paul have found an archive node issue on TW2. 

When trying to patch the archive dbs with missing blocks, He got errors because there were multiple entries in the `zkapp_account_preconditions` table that are identical except for the `id` field. 

That may have happened because there were distinct rows that became identical after the receipt_chain_hash field was patched. The duplicates are an issue because Zkapp_account_precondition.add_if_doesn't_exist function checks if a unique value exists, and return an error if there is more than one.

Probably the table should have a `UNIQUE` constraint on all the fields except id. The solution might be to find the duplicates, delete the extra ones, and fixup the references to the duplicates. 

For the "1" db, there are 80 entries with duplicates in `zkapp_account_precondition`, and in support of Paul theory of why there are duplicates, all have non-NULL entries for `receipt_chain_hash`.

### PR DESCRIPTION
**Pull request by dkijania**
Added shell script which generated two sql scripts for patching database:

- zkapp_precondition_script.sql - Removes duplicated rows in zkapp_account_precondition
- zkapp_update_body_script.sql - Updates zkapp_account_update_body table which has foreign key to zkapp_account_precondition table

Moreover i've updated zkapp_tables.sql script and added unique constraint so that whenever we bypass Ocaml code and manipulate data directly db constraint will complain

Lastly added update script which adds constraint if we want to patch existing database.

In order to fix current schemas in ITN we need to follow below steps:

- run `zkapp_duplicated_precondition/remove_duplicated_zkapp_account_precondition_rows.sh {schema}`
- run produced scripts in correct order: zkapp_precondition_script.sql, zkapp_update_body_script.sql 
- update constraint for zkapp_account_precondition table: add_constraint.sql

Closes #14777

### CODE REVIEW

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Review by ejMina226 — APPROVED** [review]
I'm not an expert on this code, but I see the priority is high.

File: scripts/archive/zkapp_duplicated_precondition/remove_duplicated_zkapp_account_precondition_rows.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+fi
+
+ARCHIVE_DB=$1
+BODY_DATA_BCK=zkapp_update_body_data_table.bck
+PRECONDITION_DATA_BCK=zkapp_precondition_table.bck
+PRECONDITION_DATA_TMP=zkapp_precondition_table.tmp
+PRECONDITION_UPDATE_SCRIPT=zkapp_precondition_script.sql
+BODY_UPDATE_SCRIPT=zkapp_update_body_script.sql
+
+echo "This script will produce two scripts which will remove duplicated zkppp account preconditions rows in archive db" "$ARCHIVE_DB"
```
Comment (shimkiv) [inline scripts/archive/zkapp_duplicated_precondition/remove_duplicated_zkapp_account_precondition_rows.sh:15]:
```suggestion
echo "This Shell script will produce two SQL scripts that will remove duplicated zkApp account preconditions rows in the archive DB" "$ARCHIVE_DB"
```

File: scripts/archive/zkapp_duplicated_precondition/remove_duplicated_zkapp_account_precondition_rows.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+do  
+    ID_TO_REPLACE=$(echo "$line" | awk -F , '{print $1}');
+    ID_TO_REMOVE=$(echo "$line" | awk -F , '{print $10}');
+    echo -n .
+    echo "$ID_TO_REPLACE $ID_TO_REMOVE" | awk '{print "UPDATE zkapp_account_update_body SET zkapp_account_precondition_id = " $1 " WHERE zkapp_account_precondition_id =" $2 ";"}' >> $BODY_UPDATE_SCRIPT
+    echo "$ID_TO_REMOVE" | awk '{print "DELETE FROM zkapp_account_precondition WHERE id=" $1 ";"}' >> $PRECONDITION_UPDATE_SCRIPT
+done
+
+echo
+echo "Scripts are ready $PRECONDITION_UPDATE_SCRIPT, $BODY_UPDATE_SCRIPT !"
```
Comment (shimkiv) [inline scripts/archive/zkapp_duplicated_precondition/remove_duplicated_zkapp_account_precondition_rows.sh:56]:
```suggestion
echo "SQL scripts are ready: $PRECONDITION_UPDATE_SCRIPT, $BODY_UPDATE_SCRIPT !"
```

**Review by shimkiv — APPROVED** [review]
LGTM, small improvements suggested.

**Comment by dkijania** [comment]
!ci-build-me

**Review by ghost-not-in-the-shell — APPROVED** [review]

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Comment by dkijania** [comment]
!ci-build-me

**Pull request merged**